### ISSUE TITLE
JIT adding new lines

### ISSUE THREAD

**Issue by rightaway** [tailwindlabs/tailwindcss#4468]
### What version of Tailwind CSS are you using?

2.1.2

### What build tool (or framework if it abstracts the build tool) are you using?

webpack

### What version of Node.js are you using?

14.16

### What browser are you using?

all

### What operating system are you using?

linux

### Reproduction repository

https://github.com/rightaway/demo

### Describe your issue

In tailwind jit it takes several `@apply` statements and puts them onto a single line. It should preserve the newlines of the original file just like tailwind always has done so that formatting the file doesn't have to become necessary in tailwind jit which it never was before.

```css
.class {
  @apply block; /* comment */
  @apply h-full; /* comment */
  @apply w-full; /* comment */
}

/* tailwind */
.class {
  display: block; /* comment */
  height: 100%; /* comment */
  width: 100% /* comment */
}

/* tailwind jit */
.class {
  display: block; /* comment */ height: 100%; /* comment */ width: 100% /* comment */
}
```

**Comment by rightaway** [comment]
Also a problem with tailwind jit when a class contains some `@apply` and some not. It's splitting it into 2 selectors when it used to be just 1 without jit.

```css
.class {
  @apply block;
  width: 100%;
  height: 100%;
}

/* tailwind */
.class {
  display: block;
  width: 100%;
  height: 100%;
}

/* tailwind jit */
.class {
  display: block;
}
.class {
  width: 100%;
  height: 100%;
}
```

**Comment by rightaway** [comment]
Here's another strange one. It splits definitions if the class list has newlines in between.

```css
.class1,
.class2,
.class3 {
  @apply w-full;
  display: block;
  @apply h-full;
}

/* tailwind jit */
.class1, .class2, .class3 {
  width: 100%;
}

.class1,
.class2,
.class3 {
  display: block;
}

.class1, .class2, .class3 {
  height: 100%;
}
```

But without newlines it's what it should be.

```css
.class1, .class2, .class3 {
  @apply w-full;
  display: block;
  @apply h-full;
}

/* tailwind jit */
.class1, .class2, .class3 {
  width: 100%;
  display: block;
  height: 100%
}
```

**Comment by adamwathan** [comment]
Hey! Some notes:

1. **Newlines being stripped when there are comments**

  I spent like 30 minutes trying to see if there was a simple fix for this and I don't really know what's happening here unfortunately, and I don't think it's worth more than the time I've spent on it to try and resolve since it's purely aesthetic and you typically minify for production anyways. I'd be open to a PR for fixing this if someone wants to work on it but unfortunately I don't think I can dedicate time to it myself, so going to close as a "won't fix".

2. **Rules being split when using `@apply`**

  I can't reproduce this locally using the code you've provided, it seems to produce the output you'd expect where everything is together in one rule.

3. **Duplicate adjacent rules not being collapsed if they have different whitespace**

  This one I actually was able to fix for you! 🥳 So will be fixed in the next release.

Thanks and sorry I can't devote the time to that first issue, and again would happily look at a PR that fixes it. PostCSS is very weird about how things are formatted when you are creating and inserting rules into the AST and I've never been able to really understand what it's doing or why 😕 If formatting for unminified builds is important to you, consider exploring PostCSS formatting plugins like this one: https://www.npmjs.com/package/style-perfectionist
